### DISCUSSION TITLE
Navigating between pages wipes out my query strings

### DISCUSSION THREAD

**Discussion by djayci** [vercel/next.js#13192] [Help]
This is my use case:

- my website buttons have a default button colour of green
- my website can receive a query string ?buttoncolour=red
- this query changes all of our buttons to red across the app

Expectation:
- the colour is persisted across the application, even when I navigate between pages
- the colour is persisted when the I refresh the page

Reality:
- the colour is persisted across the application, even when I navigate between pages
- the colour is NOT persisted when I refresh the page, as navigating between pages wipes out the query strings in the url

Any ideas? I don't really want to save this queries in cookies

**Comment by brunocrosier** [comment]
Can you create a demo repo on codesandbox or similar

**Reply by djayci** [reply]
I can, but to be honest it would be kind of a waste as that happens by default in next, without any custom code.

Using one of the Nextjs showcase websites, if you hit:
https://jobs.netflix.com/?test=true

And then click on
Read about our culture

The query string is not persisted. This is the same across all NextJS applications, as far as I'm aware.

We can then go and force those queries in you <Link> tag but then that doesn't work if you press back on your browser
